src/averageAges.js

```diff
@@ -55,10 +74,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const children = people.filter(person => {
+    return onlyWithSon
+      ? people.some(({ name }) => name === person.mother
+        && person.sex === MALE_SIGN)
+      : people.some(({ name }) => name === person.mother);
+  });
+
+  const ageDifference = children.reduce((sum, child) => {
+    return sum + (child.born - people.find(({ name }) =>
+      name === child.mother).born);
```

Comment:
To make it easier to read we can create constant value for `people.find(({ name }) =>
      name === child.mother).born`